src/utils/s3Downloader.ts

```diff
@@ -0,0 +1,58 @@
+import { Endpoint, S3 } from 'aws-sdk';
+
+const DEFAULT_PART_SIZE_BYTES = 1024 * 1024 * 8; // 8MB. Don't change this value since it will affect how etags are calculated
+
+export const DEFAULT_RETRY_POLICY = {
+  retries: 3,
+  minTimeout: 2000,
+};
+
+interface S3DownloaderOpts {
+  protocol?: string;
+  host: string;
+  port: number;
+  accessKeyId: string;
+  secretAccessKey: string;
+  computeChecksums?: boolean;
+  s3ForcePathStyle?: boolean;
+  defaultBucket: string;
+  ACL?: string;
+  ContentType?: string;
+  partSize?: number;
+}
+
+const defaultS3DownloaderOpts: Partial<S3DownloaderOpts> = {
+  protocol: 'http',
+  computeChecksums: false,
+  s3ForcePathStyle: true,
+  ACL: 'public-read',
+  ContentType: 'application/octet-stream',
+  partSize: DEFAULT_PART_SIZE_BYTES,
+};
+
+export class S3Downloader {
+  private s3: S3;
+
+  private opts: S3DownloaderOpts;
+
+  constructor(opts: S3DownloaderOpts) {
```

Comment:
this class in the original code had a lot of functionality related to S3, classes are not only useful because of its technical advantages but also because everyone can understand them. I cannot say the same of other not so common patterns as factories. I only copied the original class and removed all the unused code. I did the same for arweaveUploader too. Another advantage of having classes is testing. Remember that at first these classes were doing a lot of work, and the idea was each class to be a submodule with its own tests. Classes are a way to group functionality for a given domain, and classes are the most common way of writing code in typescript.  A few years ago I didn't like to use classes, but from my experience they are a very good way to communicate to other developers since the pattern is very common. However, I will refactor s3Downloader and arweaveUploader classes and convert them to something different.